presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformUncorrelatedInPredicateSubqueryToSemiJoin.java

```diff
@@ -106,7 +107,8 @@ public PlanNode visitApply(ApplyNode node, RewriteContext<Void> context)
                     semiJoinSymbol,
                     Optional.empty(),
                     Optional.empty(),
-                    Optional.empty()
+                    Optional.empty(),
+                    SystemSessionProperties.isLegacySemiJoin(context.get())
```

Comment:
I don't get. I don't call it "use IN as in SQL 2003 or choose IN as in My Proprietary Spec 2017".

FWIW, if the flag stays, it will be renamed to something like "support-nulls-on-lhs-of-semi-join" (default true).